### PR TITLE
feat(web-app): mini card component

### LINKED ISSUE #755: Create MiniCard component
https://gatsby-theme-carbon.vercel.app/components/MiniCard
Screen Shot 2022-05-20 at 8 07 31 AM

- [ ] Create MDX component
- [ ] Add to storybook

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #755 

#### Testing / reviewing

http://localhost:3000/mdx/mini-card
http://localhost:6006/?path=/docs/mdx-minicard--default
https://gatsby-theme-carbon.vercel.app/components/MiniCard/

### CODE REVIEW

**Review by andreancardona — COMMENTED** [review]
Not sure why the borders look slightly off to me - thoughts?
Screen Shot 2022-05-25 at 10 12 50 AM
Screen Shot 2022-05-25 at 10 12 43 AM

**Comment by alisonjoseph** [comment]
@andreancardona @francinelucca I'm getting ready to pull down and review now, but I believe because the grid adds 1px in-between, adding a border pushes that to 2px, so we will need to use box-shadow instead to fake the border.

File: services/web-app/components/mini-card/mini-card.module.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+@media screen and (prefers-reduced-motion: reduce) {
+  .card {
+    display: flex;
+    height: convert.rem(64px);
+    flex-direction: column;
+    justify-content: center;
+    background: theme.$layer-01;
+    transition: none;
+  }
+}
```
Comment (alisonjoseph) [inline services/web-app/components/mini-card/mini-card.module.scss:16]:
```suggestion
@media screen and (prefers-reduced-motion: reduce) {
  .card {
    transition: none;
  }
}
```

I think we can simplify this

File: services/web-app/components/card-group/card-group.module.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// Borders for cards at the medium breakpoint
+.card-group [class*='column']:nth-child(-n + 2) {
+  @include breakpoint.breakpoint('md') {
+    border-top: 0 solid transparent;
+  }
+}
+
+.card-group [class*='column']:nth-child(even) {
+  @include breakpoint.breakpoint('md') {
+    border-left: 1px solid theme.$border-subtle-01;
```
Comment (alisonjoseph) [inline services/web-app/components/card-group/card-group.module.scss:22]:
```suggestion
    box-shadow: -1px 0 0 0 theme.$border-subtle;
```
this should fix the extra pixel issue @andreancardona was showing

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
Looking good! and good call on creating the card group component. 

Do we want to wrap the ResourceCard examples with this as a preferred way to do that vs. the current classname?

File: services/web-app/components/card-group/card-group.module.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
+@use '@carbon/react/scss/theme' as theme;
+@use '@carbon/react/scss/breakpoint' as breakpoint;
+// Sets gray border on the top of cards at the smallest breakpoint and removes the margin set by the <Column> in the Grid component.
+.card-group [class*='column'] {
+  border-top: 1px solid theme.$border-subtle-01;
```
Comment (alisonjoseph) [inline services/web-app/components/card-group/card-group.module.scss:5]:
```suggestion
  border-top: 1px solid theme.$border-subtle1;
```

File: services/web-app/components/card-group/card-group.module.scss (hunk 2 above)
Comment (francinelucca) [reply services/web-app/components/card-group/card-group.module.scss]: thanks!!

**Comment by francinelucca** [comment]
> Looking good! and good call on creating the card group component.
> 
> Do we want to wrap the ResourceCard examples with this as a preferred way to do that vs. the current classname?

wrapped the resource card mdx and story in a card group and lgtm!

**Review by alisonjoseph — APPROVED** [review]

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
Just realized we need to support the mdx icon compoennt as a child here and styles are a little off. 

For example.
```

<CardGroup>
  <MiniCard title="React tutorial" href="">
    <MdxIcon name="react" />
  </MiniCard>
  ....

```
Screen Shot 2022-05-27 at 8 37 39 AM

File: services/web-app/components/mini-card/mini-card.module.scss [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  justify-content: space-between;
+  padding-right: spacing.$spacing-05;
+  padding-left: spacing.$spacing-05;
+}
+
+.title {
+  @include type.type-style('body-short-01');
+}
+
+.image {
```
Comment (alisonjoseph) [inline services/web-app/components/mini-card/mini-card.module.scss:38]:
```suggestion
.image image {
```
This should fix the image alignment being off

Comment (alisonjoseph) [inline services/web-app/components/mini-card/mini-card.module.scss:38]:
```suggestion
.image,
.image :global([class*='mdx-icon']) 
```

**Comment by alisonjoseph** [comment]
Found something else weird, if I add a relative path in the href they look like this.
```

<MiniCard title="React tutorial" href="/assets/carbon-components-react">
    <MdxIcon name="react" />
  </MiniCard>
```

Screen Shot 2022-05-27 at 9 16 26 AM

File: services/web-app/components/mini-card/mini-card.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { Column, Link as CarbonLink } from '@carbon/react'
```
Comment (alisonjoseph) [inline services/web-app/components/mini-card/mini-card.js:7]:
```suggestion
import { Column } from '@carbon/react'
```

File: services/web-app/components/mini-card/mini-card.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    isLink = href.charAt(0) === '/'
+  }
+
+  return (
+    <Column md={4} lg={4} sm={4} {...rest}>
+      {isLink && (
+        <Link href={href} passHref>
+          <CarbonLink to={href} className={'cds--tile--clickable'} {...linkProps}>
+            {cardContent}
+          </CarbonLink>
```
Comment (alisonjoseph) [inline services/web-app/components/mini-card/mini-card.js:52]:
```suggestion
          <a to={href} className={'cds--tile--clickable'} {...linkProps}>
            {cardContent}
          </a>
```

File: services/web-app/components/mini-card/mini-card.module.scss (hunk 4 above)
Comment (francinelucca) [reply services/web-app/components/mini-card/mini-card.module.scss]: thanks 🙏

File: services/web-app/components/mini-card/mini-card.js (hunk 6 above)
Comment (francinelucca) [reply services/web-app/components/mini-card/mini-card.js:52]: thanks 🙏

**Review by alisonjoseph — APPROVED** [review]
🥳

**Review by andreancardona — APPROVED** [review]
🔥 looks good!

**Pull request merged**